rsts/deployment/cluster_config/performance.rst

```diff
@@ -13,16 +13,49 @@ The video below contains an overview of the Flyte architecture, what is meant by
+It is designed as a `Kubernetes Controller <https://kubernetes.io/docs/concepts/architecture/controller/>`_, where the desired state is specified as a FlyteWorkflow `Custom Resource <https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/>`_.
+
+One of the base assumptions of FlytePropeller is that every workflow is independent and can be executed by a completely distinct process, without a need for communication with other processes. On the flip-side, one workflow is tracks the dependencies using a DAG structure and hence constantly needs synchronization.
+Currently each workflow is executed by exactly one process and one thread at a given time. Each workflow is constantly evaluated in an event-loop and uses the specification to cause an effect on the K8s cluster or remote services. The actual effect is actually caused by FlytePropeller plugins.
+
 FlytePropeller has a lot of knobs that can be tweaked for performance. The default configuration is good enough for small to medium sized installations of Flyte, that are running about 500 workflows concurrently with no noticeable overhead. In the case when the number of workflows increases,
 FlytePropeller will automatically slow down, without losing correctness.
 
-Here are signs of slowdown,
+Signs of slowdown
+------------------
 
-#. Round Latency for each workflow increases
-#. Transition latency increases
-#. Workflows take longer to start
+.. list-table:: Important metrics to look out for in Prometheus dashboard
+   :widths: 25 25 50 100
+   :header-rows: 1
 
-This is usually because the number of threads in FlytePropeller are not enough to keep up with the number of workflows. This can be resolved by adjusting FlytePropeller config specified `here <https://pkg.go.dev/github.com/flyteorg/flytepropeller@v0.10.3/pkg/controller/config>`_.
+   * - Metric
+     - Dashboard
+     - Alerting factor
+     - Effect
+   * - flyte:propeller:all:free_workers_count
+     - Flyte Propeller Dashboard
+     - Number of free-workers in all flytepropeller instances are very low.
+     - This will increase overall latency for Each workflow propagation
+   * - sum(rate(flyte:propeller:all:round:raw_ms[5m])) by (wf)
+     - Flyte Propeller Dashboard
+     - Round Latency for Each workflow increases
+     - Flyte propeller is taking more time to process each workflow
+   * - sum(rate(flyte:propeller:all:main_depth[5m]))
+     - Flyte Propeller Dashboard
+     - Workflows take longer to start or slow
+     - The processing queue depth is long and is taking long to drain
+
+For each of the metrics above you can dig deeper into the cause for this spike in latency. All of them are mostly related to one latency and should be correlated - ``The Round latency!``.
+The round-latency is the time it takes for FlytePropeller to traverse through one workflow. To understand this, you have to understand the :ref:`divedeep-execution-timeline`. Once you understand this, continue reading on.
```

Comment:
to traverse through one workflow -> to perform a single iteration of workflow evaluation